ext/fg/js/frontend.js

```diff
@@ -262,7 +371,97 @@ class Frontend {
 
         this.textSourceLast = null;
     }
+
+    getPrimaryTouch(touchList) {
+        return touchList.length > 0 ? touchList[0] : null;
+    }
+
+    getIndexOfTouch(touchList, identifier) {
+        for (let i = 0, ii = touchList.length; i < ii; ++i) {
```

Comment:
Use `for` `of` loop here instead of manually indexing.

Reply:
```for``` ```in``` since the index what is returned.